### DISCUSSION TITLE
ICS20: Memo field

### DISCUSSION THREAD

**Discussion by sunnya97** [cosmos/ibc#727] [Ideas]
Thoughts on adding an additional "memo" field to the ICS20 `FungibleTokenPacketData` struct?  Often times, integrators such as centralized exchanges use memo fields to carry information about which depositing account to credit with tokens.  By adding a "memo" field to ICS20 transactions, then users would be able to deposit tokens directly to a centralized exchange via ICS20 transfers.  For example, a user could create an ICS20 transfer of ATOMs from Osmosis to Coinbase's Hub address with their user memo directly, instead of having to first ICS20 transfer to Cosmos Hub, and then do a normal transfer message.

**Comment by alexanderbez** [comment]
Love the idea

**Comment by colin-axner** [comment]
I think most folks are in favor, see #584 

The tricky part is getting existing channels to send/decode packets with the additional memo field. However, now that the channel upgradeability spec has been merged, this is actually fairly straightforward. We would just upgrade existing channels to a `ics20-2` version which would decode to an extended definition of `FungibleTokenPacketData`

**Comment by colin-axner** [comment]
see https://github.com/cosmos/ibc/pull/842
